skimage/future/graph/graph_cut.py

```diff
@@ -138,6 +138,82 @@ def cut_normalized(labels, rag, thresh=0.001, num_cuts=10, in_place=True,
     return map_array[labels]
 
 
+def cut_normalized_gen(labels, rag, thresh=[0.001], num_cuts=10, in_place=True,
+                   max_edge=1.0):
+    """Perform Normalized Graph cut on the Region Adjacency Graph.
+
+    Given an image's labels and its similarity RAG, recursively perform
+    a 2-way normalized cut on it. All nodes belonging to a subgraph
+    that cannot be cut further are assigned a unique label in the
+    output. Appending a new value to the thresh list and calling 
+    next on this generator returns a new array of labels updated 
+    for the new thresh value. 
+
+    Parameters
+    ----------
+    labels : ndarray
+        The array of labels.
+    rag : RAG
+        The region adjacency graph.
+    thresh : list
+        List of threshold values to compute. A subgraph won't be further 
+        subdivided if the value of the N-cut exceeds a threshold value. 
+    num_cuts : int
+        The number or N-cuts to perform before determining the optimal one.
+    in_place : bool
+        If set, modifies `rag` in place. For each node `n` the function will
+        set a new attribute ``rag.node[n]['ncut label']``.
+    max_edge : float, optional
+        The maximum possible value of an edge in the RAG. This corresponds to
+        an edge between identical regions. This is used to put self
+        edges in the RAG.
+
+    Returns
+    -------
+    out : ndarray
+        The new labeled array.
+
+    Examples
+    --------
+    >>> from skimage import data, segmentation
+    >>> from skimage.future import graph
+    >>> img = data.astronaut()
+    >>> labels = segmentation.slic(img, compactness=30, n_segments=400)
+    >>> rag = graph.rag_mean_color(img, labels, mode='similarity')
+    >>> 
+    >>> thresh = [1e-5]
```

Comment:
Do these values always get pushed one by one?  In that case, this may be a good case for https://docs.python.org/2/reference/expressions.html#generator.send

Reply:
I don't think I've used that, but reading the docs I think this would be a good case. It felt a little hacky to have the user pass values by appending to a list and this should get around that. I'll make the changes.